aws_signing_helper/pkcs11_signer.go

```diff
@@ -77,14 +206,45 @@ fail:
 	return nil, 0, err
 }
 
-// Gets all certificates within the PKCS #11 session
-func getCerts(module *pkcs11.Ctx, session pkcs11.SessionHandle) (certs []*x509.Certificate, err error) {
+// Convert the object-related fields in a URI to pkcs11.Attributes for FindObjectsInit()
+func getFindTemplate(uri *pkcs11uri.Pkcs11URI, class uint) (template []*pkcs11.Attribute) {
+	var v string
+	var ok bool
+
+	template = append(template, pkcs11.NewAttribute(pkcs11.CKA_CLASS, class))
+
+	if uri == nil {
+		return template
+	}
+
+	v, ok = uri.GetPathAttribute("object", false)
+	if ok {
+		template = append(template, pkcs11.NewAttribute(pkcs11.CKA_LABEL, v))
+	}
+	v, ok = uri.GetPathAttribute("id", false)
+	if ok {
+		template = append(template, pkcs11.NewAttribute(pkcs11.CKA_ID, v))
+	}
+	return template
```

Comment:
Are `CKA_ID` and `CKA_LABEL` the only two attributes that should be used to match a certificate?

Reply:
Yes, they're the only object-specific fields that exist in RFC7512.